eldritch/expansions/clifftown/test_clifftown_abilities.py

```diff
@@ -0,0 +1,88 @@
+
+from eldritch.test_events import EventTest
+from eldritch.expansions.clifftown import abilities
+from eldritch import characters
+from eldritch import events
+from eldritch import monsters
+from eldritch.items import unique
+
+
+class TestUrchinAbilities(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.elder_sign = unique.ElderSign(0)
+    self.minor = abilities.Minor()
+    self.char.possessions.extend(
+        [
+            abilities.Streetwise(), abilities.BlessedIsTheChild(),
+            self.minor, self.elder_sign,
+        ]
+    )
+
+  def testEvadeInStreets(self):
+    self.char.place = self.state.places["Northside"]
+    cultist = monsters.Cultist()
+    combat = events.Combat(self.char, cultist)
+    self.state.event_stack.append(combat)
+    fight_or_flee = self.resolve_to_choice(events.FightOrEvadeChoice)
+    fight_or_flee.resolve(self.state, "Flee")
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=1)) as dice:
+      self.resolve_until_done()
+      self.assertEqual(dice.call_count, 0)
+
+  def testEvadeInLocation(self):
+    self.char.speed_sneak_slider = 0
+    cultist = monsters.Cultist()
+    combat = events.Combat(self.char, cultist)
+    self.state.event_stack.append(combat)
+    fight_or_flee = self.resolve_to_choice(events.FightOrEvadeChoice)
+    fight_or_flee.resolve(self.state, "Flee")
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)) as dice:
+      self.resolve_until_done()
+      self.assertEqual(dice.call_count, 1)
+
+  def testCannotCurse(self):
+    curse = events.Curse(self.char)
+    self.state.event_stack.append(curse)
+    self.resolve_until_done()
+
+    self.assertEqual(self.char.bless_curse, 0)
+    self.assertTrue(curse.cancelled)
+
+    self.state.event_stack.append(events.DiscardSpecific(self.char, [self.elder_sign]))
+    self.resolve_until_done()
+
+    curse = events.Curse(self.char)
+    self.state.event_stack.append(curse)
+    self.resolve_until_done()
+
+    self.assertEqual(self.char.bless_curse, -1)
+    self.assertFalse(curse.cancelled)
+
+  def testCannotBeArrested(self):
+    arrested = events.Arrested(self.char)
+    self.state.event_stack.append(arrested)
+    self.resolve_until_done()
+
+    self.assertIsNone(self.char.arrested_until)
+    self.assertTrue(arrested.cancelled)
+
+    self.state.event_stack.append(events.DiscardSpecific(self.char, [self.elder_sign]))
+    self.resolve_until_done()
+
+    arrested = events.Arrested(self.char)
+    self.state.event_stack.append(arrested)
+    self.resolve_until_done()
+
+    self.assertIsNotNone(self.char.arrested_until)
+    self.assertFalse(arrested.cancelled)
+
+  def testMinorCredit(self):
+    self.char.place = self.state.places["Bank"]
+    self.advance_turn(0, "encounter")
+    self.resolve_until_done()
+
+    self.char.possessions.remove(self.minor)
+    self.advance_turn(1, "encounter")
+    self.resolve_until_done()
```

Comment:
This test isn't working as I anticipated... do the FixedEncounters not get loaded when in test mode?

Reply:
Can you define not working as anticipated? See other comments about can_take vs can_get for my theory on why this may not be working as expected.

Reply:
When the DummyCharacter is no longer a minor (happy birthday?), I was anticipating them to be offered a choice to either take a loan or have a regular encounter. But the fact that they can resolve_until_done suggests I'm misunderstanding something.

Reply:
Oh, you're correct - fixed encounters do not get initialized in test mode. You'll need to add these three lines https://github.com/gummyboars/lunboks/blob/master/eldritch/test_game.py#L55-L57
to your setUp() to make it work.